pkg/controller/replicaset/replica_set.go

```diff
@@ -327,22 +327,22 @@ func (rsc *ReplicaSetController) updatePod(old, cur interface{}) {
 	curPod := cur.(*api.Pod)
 	oldPod := old.(*api.Pod)
 	glog.V(4).Infof("Pod %s updated %+v -> %+v.", curPod.Name, oldPod, curPod)
-	rs := rsc.getPodReplicaSet(curPod)
-	if rs == nil {
-		return
-	}
-
 	if curPod.DeletionTimestamp != nil {
 		// when a pod is deleted gracefully it's deletion timestamp is first modified to reflect a grace period,
 		// and after such time has passed, the kubelet actually deletes it from the store. We receive an update
 		// for modification of the deletion timestamp and expect an rs to create more replicas asap, not wait
 		// until the kubelet actually deletes the pod. This is different from the Phase of a pod changing, because
 		// an rs never initiates a phase change, and so is never asleep waiting for the same.
 		rsc.deletePod(curPod)
+		if !reflect.DeepEqual(curPod.Labels, oldPod.Labels) {
```

Comment:
nit: please extract the deepequal into a local var with a name like labelsChanged since we're doing it twice, and add a comment about why it's safe to just invoke deletePod with the oldPod without checking delete timestamp (i.e we can never have a curPod without timestmap and oldPod with).